### DISCUSSION TITLE
How to install the dependencies in a specific workspace

### DISCUSSION THREAD

**Discussion by amirhhashemi** [yarnpkg/berry#3558] [Q&A] [answered]
I'm new to monorepos with yarn. I have a monorepo with a folder structure like below:
```
project
        package.json
        yarn.lock
        api
            package.json
            Dockerfile
        client
            package.json
```
I want to just install the dependencies related to the api. I tried `yarn workspace api install` and `yarn install focus` but none of them creates a `node_modules` folder in the workspace folder (I'm using yarn v3.0.2 with the node-modules linker). they always create `node_modules` in the project directory with all dependencies:
```
project
        node_modules
        api
        client
```
I expected something like this:
```
project
        api
                node_modules
        client
```
btw, the reason I want to do that is that I'm trying to dockerize the api. I'm wondering what command should I run in Dockerfile?

can anyone help me?

**Comment by saini-g** [comment]
Have you tried adding the api workspace to the `pnpIgnorePatterns` as mentioned in [this recipe](https://yarnpkg.com/getting-started/recipes#hybrid-pnp--node_modules-mono-repo)

Please do let me know if the docker setup works out for you, I am also looking to do a similar setup

**Reply by amirhhashemi** [reply]
Sorry, this is an old question, and I forgot exactly what I wanted at the time and how I solved it. But I tried the instructions in the link you provided, and it works. This is my new folder structure:
```
project
        package.json
        yarn.lock
        .yarnrc.yml     // new
        api
            package.json
            .yarnrc.yml       // new
            yarn.lock      // new
            Dockerfile
        client
            package.json
```
This is the content of api/.yarnrc.yml :
```
nodeLinker: node-modules
```
and this is the content of the `.yarnrc.yml` file in the root:
```
pnpIgnorePatterns:
  - ./**
```
By running `yarn` in the `api` folder, yarn creates an `api/node_modules` folder. Technically this Dockerfile should work in this situation:
```
FROM node:16-alpine

WORKDIR /app

COPY api/package.json ./
COPY yarn.lock ./

RUN yarn install

COPY api/ ./

EXPOSE 3000

CMD ["yarn", "start"]
```
Note that I build the docker image in the context of the root directory using this command:
```
docker build -f api/Dockerfile .
```

I forgot how I dealt with my problem at the time, but this sounds like a good solution.

**Comment by TrySound** [comment, marked as answer]
Don't forget to install workspaces-tools plugin
```
yarn workspaces focus workspace-name
```

**Reply by amirhhashemi** [reply]
For some reason, it doesn't work. It installs all dependencies in all workspaces and creates the `node_modules` folder in the root instead of the `api` folder. I think `yarn workspaces ...` installs dependencies in a specific workspace, but I'm not sure if it creates the `node_module` folder in the workspace too. 

Btw the @saini-g 's solution seems to work for me.

**Reply by larixer** [reply]
@ahhshm If you want `node_modules` to be contained inside `workspace-name` you should rather do `YARN_NM_HOISTING_LIMITS=workspaces yarn workspaces focus workspace-name`

**Reply by amirhhashemi** [reply]
Thank you, @larixer. Exactly what I wanted.
